source_control/gh_keys.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2013, André Paramés <git@andreparames.com>
+# Based on the Git module by Leonardo Comelli <leonardo.comelli@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import base64
+
+DOCUMENTATION = '''
+---
+module: gh_keys
+version_added: 1.9 
+short_description: Manages github ssh keys.
+description:
+    - The module manages the ssh key for a specific user through Github API v3.
+options:
+  action:
```

Comment:
action is imperative, we favor declaritive notation: state: present, absent (list/get can be removed in favor of always returning the keys after operation).